### PR TITLE
【Prim】Add cinn tests for dropout decomposite

### PR DESCRIPTION
**Pull request by cxxly**
### PR types

Others
### PR changes

Others
### Describe

- Add cinn tests for dropout decomposite

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/dygraph_to_static/test_cinn_prim_dropout.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+            loss.backward()
+            fwd += loss.numpy()
+            rev += self.x.grad.sum().numpy()
+            sgd.step()
+            self.x.clear_gradient()
+        return fwd, rev, net
+
+    def check_prim(self, net):
+        ops = [op.type for op in net.forward.main_program.block(0).ops]
+        # Ensure that softmax is splitted into small ops
```
Comment (cyber-pioneer) [inline python/paddle/fluid/tests/unittests/dygraph_to_static/test_cinn_prim_dropout.py]: dropout

File: python/paddle/fluid/tests/unittests/dygraph_to_static/test_cinn_prim_dropout.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        self.assertTrue('dropout' not in ops)
+
+    def test_cinn_prim_forward(self):
+        paddle.seed(1999)
+        dy_fwd, dy_rev, _ = self.train(use_prim=False, use_cinn=False)
+        paddle.seed(1999)
```
Comment (cyber-pioneer) [inline python/paddle/fluid/tests/unittests/dygraph_to_static/test_cinn_prim_dropout.py:72]: 设置2次和设置一次有差异么？

File: python/paddle/fluid/tests/unittests/dygraph_to_static/test_cinn_prim_dropout.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+class TestPrimForwardAndBackward(unittest.TestCase):
+    """
+    This case only tests prim_forward + to_static + cinn. Thus we need to
+    set this flag as False to avoid prim_backward.
+    core.set_prim_backward(False)
+    """
+
+    def setUp(self):
+        self.x = paddle.ones([100, 100, 10])
```
Comment (cyber-pioneer) [inline python/paddle/fluid/tests/unittests/dygraph_to_static/test_cinn_prim_dropout.py:41]: 全1的输入，输出的精度阈值是否更高？

File: python/paddle/fluid/tests/unittests/dygraph_to_static/test_cinn_prim_dropout.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+    def forward(self, x):
+        return F.dropout(self.fc(x))
+
+
+class TestPrimForwardAndBackward(unittest.TestCase):
+    """
+    This case only tests prim_forward + to_static + cinn. Thus we need to
+    set this flag as False to avoid prim_backward.
+    core.set_prim_backward(False)
```
Comment (cyber-pioneer) [inline python/paddle/fluid/tests/unittests/dygraph_to_static/test_cinn_prim_dropout.py]: remove

File: python/paddle/fluid/tests/unittests/dygraph_to_static/test_cinn_prim_dropout.py [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+        self.x = paddle.ones([100, 100, 10])
+        self.x.stop_gradient = False
+
+    def train(self, use_prim=True, use_cinn=False):
+        net = PrimeNet()
+        sgd = paddle.optimizer.SGD(
+            learning_rate=0.1, parameters=net.parameters()
+        )
+
+        core._set_prim_forward_enabled(use_prim)
```
Comment (cyber-pioneer) [inline python/paddle/fluid/tests/unittests/dygraph_to_static/test_cinn_prim_dropout.py]: prim all

File: python/paddle/fluid/tests/unittests/dygraph_to_static/test_cinn_prim_dropout.py (hunk 2 above)
Comment (cxxly) [reply python/paddle/fluid/tests/unittests/dygraph_to_static/test_cinn_prim_dropout.py:72]: 有，采样算法是从一个初始seed开始迭代，设置一次，采样两次，相当于从一个初始seed迭代两次；这里要保证初始seed一致

File: python/paddle/fluid/tests/unittests/dygraph_to_static/test_cinn_prim_dropout.py (hunk 3 above)
Comment (cxxly) [reply python/paddle/fluid/tests/unittests/dygraph_to_static/test_cinn_prim_dropout.py:41]: 更稳定，避免单测随机失败

File: python/paddle/fluid/tests/unittests/dygraph_to_static/test_cinn_prim_dropout.py (hunk 4 above)
Comment (cxxly) [reply python/paddle/fluid/tests/unittests/dygraph_to_static/test_cinn_prim_dropout.py]: done

File: python/paddle/fluid/tests/unittests/dygraph_to_static/test_cinn_prim_dropout.py (hunk 5 above)
Comment (cxxly) [reply python/paddle/fluid/tests/unittests/dygraph_to_static/test_cinn_prim_dropout.py]: done

File: python/paddle/fluid/tests/unittests/dygraph_to_static/test_cinn_prim_dropout.py (hunk 1 above)
Comment (cxxly) [reply python/paddle/fluid/tests/unittests/dygraph_to_static/test_cinn_prim_dropout.py]: done

**Comment by cyber-pioneer** [comment]
LGTM
